benchmarks/linear-elastic-plate-with-hole/workflow_creation_guide.md

```diff
@@ -0,0 +1,930 @@
+        --output_metrics_file "solution_metrics_${configuration}.json"
+    """
+}
+
+workflow tool1_workflow {
+    take: 
+    mesh_data // tuple(configuration, parameters, mesh) 
+    result_dir
+
+    main:
+    params.result_dir = result_dir
+    run_sim_script = Channel.value(file('tool1/run_tool1_simulation.py'))
+    output_process_run_simulation = run_simulation(run_sim_script, mesh_data)
+
+    emit:
+    output_process_run_simulation
+}
+```
+
+### Step 6: Implement Simulation Scripts
+
+Each tool needs a standardized simulation script:
+
+```python
+import json
+import numpy as np
+from argparse import ArgumentParser
+from pathlib import Path
+from pint import UnitRegistry
+
+def run_simulation(parameter_file: str, mesh_file: str, 
+                  solution_file_zip: str, metrics_file: str) -> None:
+    ureg = UnitRegistry()
+    
+    # Load parameters
+    with open(parameter_file) as f:
+        parameters = json.load(f)
+    
+    # Extract parameters with unit conversion
+    E = ureg.Quantity(parameters["young-modulus"]["value"], 
+                     parameters["young-modulus"]["unit"]).to_base_units().magnitude
+    nu = ureg.Quantity(parameters["poisson-ratio"]["value"], 
+                      parameters["poisson-ratio"]["unit"]).to_base_units().magnitude
+    
+    # Load mesh and run simulation
+    # ... tool-specific simulation code ...
+    
+    # Calculate metrics
+    max_stress = float(np.max(stress_field))
+    max_displacement = float(np.max(displacement_field))
+    
+    # Save metrics
+    metrics = {
+        "max_von_mises_stress_nodes": max_stress,
+        "max_displacement": max_displacement,
+        "total_elements": int(num_elements),
+        "total_nodes": int(num_nodes),
+        "solver_iterations": int(iterations),
+        "computation_time": float(computation_time)
+    }
+    
+    with open(metrics_file, "w") as f:
+        json.dump(metrics, f, indent=4)
+    
+    # Save solution field data as ZIP
+    import zipfile
+    with zipfile.ZipFile(solution_file_zip, 'w') as zf:
+        # Add VTK files, displacement fields, stress fields, etc.
+        for file_path in solution_files:
+            zf.write(file_path, Path(file_path).name)
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="Run simulation with specified tool")
+    parser.add_argument("--input_parameter_file", required=True)
+    parser.add_argument("--input_mesh_file", required=True)
+    parser.add_argument("--output_solution_file_zip", required=True)
+    parser.add_argument("--output_metrics_file", required=True)
+    args = parser.parse_args()
+    
+    run_simulation(args.input_parameter_file, args.input_mesh_file,
+                   args.output_solution_file_zip, args.output_metrics_file)
+```
+
+### Step 7: Implement Results Summarization
+
+Create `summarise_results.py`:
+
+```python
+from argparse import ArgumentParser
+import json
+
+def create_summary(configurations: list[str],
```

Comment:
this is actually also identical in all benchmarks, so maybe we move all that to a general folder (under benchmarks) that includes all the general files, also the nextflow.config